extensions/windows-event-log/wel/JSONUtils.cpp

```diff
@@ -135,28 +157,37 @@ rapidjson::Document toJSONImpl(const pugi::xml_node& root, bool flatten) {
 
   {
     auto eventData_xml = event_xml.child("EventData");
-    // create EventData subarray even if flatten requested
-    doc.AddMember("EventData", rapidjson::kArrayType, doc.GetAllocator());
-    for (const auto& data : eventData_xml.children()) {
-      auto name_attr = data.attribute("Name");
-      rapidjson::Value item(rapidjson::kObjectType);
-      item.AddMember("Name", rapidjson::StringRef(name_attr.value()), doc.GetAllocator());
-      item.AddMember("Content", rapidjson::StringRef(data.text().get()), doc.GetAllocator());
-      item.AddMember("Type", rapidjson::StringRef(data.name()), doc.GetAllocator());
-      // we need to query EventData because a reference to it wouldn't be stable, as we
-      // possibly add members to its parent which could result in reallocation
-      doc["EventData"].PushBack(item, doc.GetAllocator());
-      // check collision
-      if (flatten && !name_attr.empty() && !doc.HasMember(name_attr.value())) {
-        doc.AddMember(rapidjson::StringRef(name_attr.value()), rapidjson::StringRef(data.text().get()), doc.GetAllocator());
+    if (flatten) {
+      for (const auto& event_data_child : eventData_xml.children()) {
+        std::string key = "EventData";
+        if (auto name_attr = event_data_child.attribute("Name").value(); strlen(name_attr)) {
+          key = utils::StringUtils::join_pack(key, ".", name_attr);
+        }
+
+        if (auto type = event_data_child.name(); strlen(type) > 0) {
+          key = utils::StringUtils::join_pack(key, ".", type);
+        }
+
+        doc.AddMember(rapidjson::Value(createUniqueKey(key, doc), doc.GetAllocator()).Move(), rapidjson::StringRef(event_data_child.text().get()), doc.GetAllocator());
+      }
+    } else {
+      auto& event_data = doc.AddMember("EventData", rapidjson::kArrayType, doc.GetAllocator());
+      for (const auto& event_data_child : eventData_xml.children()) {
+        auto name_attr = event_data_child.attribute("Name");
+        rapidjson::Value item(rapidjson::kObjectType);
+        item.AddMember("Name", rapidjson::StringRef(name_attr.value()), doc.GetAllocator());
+        item.AddMember("Content", rapidjson::StringRef(event_data_child.text().get()), doc.GetAllocator());
+        item.AddMember("Type", rapidjson::StringRef(event_data_child.name()), doc.GetAllocator());
+        doc["EventData"].PushBack(item, doc.GetAllocator());
```

Comment:
This would lose the type information (which might be useful), we could do of course something similar to the flat output so something like this?
```
"EventData": {
    "Foobar.Text": "Lorem ipsum"
}
```
But since this is only a bugfix for the flat output type I wouldnt change the simple format in this PR maybe we could file a Jira ticket and explore the possibilities there?